org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintUiPlugin.java

```diff
@@ -245,21 +246,25 @@ private static void addListenerToAllPages(IWorkbenchWindow window) {
     @Override
     public IStatus runInUIThread(IProgressMonitor monitor) {
 
-      JobUtils.scheduleAnalysisOfOpenFiles((IProject) null, TriggerType.STARTUP);
+      JobUtils.scheduleAnalysisOfOpenFiles((ISonarLintProject) null, TriggerType.STARTUP);
 
       return Status.OK_STATUS;
     }
 
   }
 
   @CheckForNull
-  public static IFile findCurrentEditedFile() {
+  public static ISonarLintFile findCurrentEditedFile() {
     IEditorPart editor = findActiveEditor();
     if (editor == null) {
       return null;
     }
     // note: the cast is necessary for e43 and e44
-    return (IFile) editor.getEditorInput().getAdapter(IFile.class);
+    IFile file = (IFile) editor.getEditorInput().getAdapter(IFile.class);
+    if (file != null) {
+      return (ISonarLintFile) file.getAdapter(ISonarLintFile.class);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "ISonarLintFile". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1905)